File: threedollar-domain/src/main/java/com/depromeet/threedollar/domain/user/domain/storedelete/StoreDeleteRequestCreator.java
package com.depromeet.threedollar.domain.user.domain.storedelete;

import com.depromeet.threedollar.common.docs.ObjectMother;
import com.depromeet.threedollar.domain.user.domain.store.Store;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@ObjectMother
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class StoreDeleteRequestCreator {

    public static StoreDeleteRequest create(Store store, Long userId, DeleteReasonType type) {
        return StoreDeleteRequest.of(store, userId, type);
    }

}

File: threedollar-domain/src/testFixtures/java/com/depromeet/threedollar/domain/user/domain/storedelete/StoreDeleteRequestCreator.kt
package com.depromeet.threedollar.domain.user.domain.storedelete

import com.depromeet.threedollar.domain.user.domain.TestFixture
import com.depromeet.threedollar.domain.user.domain.store.Store

@TestFixture
object StoreDeleteRequestCreator {

    @JvmStatic
    fun create(
        store: Store,
        userId: Long,
        reasonType: DeleteReasonType
    ): StoreDeleteRequest {
        return StoreDeleteRequest.builder()
            .store(store)
            .userId(userId)
            .reason(reasonType)
            .build()
    }

}
